framework/db/mysql/QueryBuilder.php

```diff
@@ -390,7 +390,15 @@ private function getColumnDefinition($table, $column)
      */
     private function supportsFractionalSeconds()
     {
-        $version = $this->db->getSlavePdo()->getAttribute(\PDO::ATTR_SERVER_VERSION);
+        // use cache to prevent open mysql connection
+        // https://github.com/yiisoft/yii2/issues/13749#issuecomment-481657224
+        $key = [__METHOD__, $this->db->dsn];
+        $version = isset(\Yii::$app->cache) ? \Yii::$app->cache->get($key) : null;
+        if( !$version ) {
+            $version = $this->db->getSlavePdo()->getAttribute(\PDO::ATTR_SERVER_VERSION);
+            isset(\Yii::$app->cache) ? \Yii::$app->cache->set($key, $version) : null;
```

Comment:
Please use regular `if` here - ternary operator is unreadable and hides complexity.